src/sentry/static/sentry/app/views/eventsV2/utils.tsx

```diff
@@ -336,3 +337,18 @@ export function getPrebuiltQueries(organization: Organization) {
 
   return views;
 }
+
+export function decodeScalar(
```

Comment:
Should the various decode methods be moved here or somewhere more general? As we convert more code to typescript we're going to have more copies of these decode functions floating about.

Reply:
They definitely should. I've been meaning to want to do this; but I prefer to do this with a another PR that handles general refactoring (they're usually 1k+ LOC diff changes on my part).